.github/workflows/benchmark.yml

```diff
@@ -0,0 +1,175 @@
+# This workflow is the main workflow for regenerating the benchmarks data needed for Bowtie's UI.
+# It runs all benchmarks over Bowtie's supported implementations, publishing the benchmark reports for use in the frontend.
+name: Collect New Benchmark Results
+
+on:
+  workflow_dispatch:
+  schedule:
+    # Every Monday at 08:00 UTC
+    - cron: "0 8 * * 1"
+
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+jobs:
+  dialects:
+    runs-on: ubuntu-latest
+    outputs:
+      dialects: ${{ steps.dialects-matrix.outputs.dialects }}
+    steps:
+      - uses: actions/checkout@v4
+      - name: Collect supported dialects
+        id: dialects-matrix
+        run: |
+          printf 'dialects=%s\n' "$(jq -c '[.[].shortName]' data/dialects.json)" >> $GITHUB_OUTPUT
+
+  benchmark_files:
+    needs: dialects
+    runs-on: ubuntu-latest
+    outputs:
+      dialect_benchmarks: ${{ steps.benchmarks.outputs.dialect_benchmarks }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install Bowtie
+        uses: ./
+
+      - name: Collect Benchmark Files
+        id: benchmarks
+        run: |
+          results=()
+          dialects='${{ needs.dialects.outputs.dialects }}'
+          dialects=$(echo $dialects | jq -r '.[]')
+          results=()
+
+          for dialect in $dialects; do
+            output=$(bowtie filter-benchmarks -D "$dialect")
+
+            if [ -n "$output" ]; then
+              while IFS= read -r line; do
+                json_result=$(jq -nc --arg p "$dialect" --arg o "$line" '{ dialect: $p, benchmark: $o }')
+                results+=("$json_result")
+              done <<< "$output"
+            fi
+          done
+          final_json="$(jq -sc '.' <<< "${results[@]}")"
+          final_json=$(echo "$final_json" | jq -c '{ "include": . }')
+          echo $final_json >> $GITHUB_STEP_SUMMARY
+          echo "dialect_benchmarks=$final_json" >> $GITHUB_OUTPUT
+
+  run_benchmarks:
+    needs: benchmark_files
+    runs-on: ubuntu-latest
+    timeout-minutes: 720
+    strategy:
+      fail-fast: false
+      matrix: ${{ fromJson(needs.benchmark_files.outputs.dialect_benchmarks) }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install Bowtie
+        uses: ./
+
+      - name: Install pyperf dependency
+        run: |
+          python -m pip install pyperf
+
+      - name: Generate Benchmark Report
+        run: |
+          bowtie perf $(bowtie filter-implementations | sed 's/^/-i /') -b ${{ matrix.benchmark }} -D ${{ matrix.dialect }} -q > ${{ matrix.benchmark }}-file.json
+
+      - name: Store Unique Id for distinction
+        run: echo "unique_id=$(uuidgen)" >> $GITHUB_ENV
```

Comment:
I'm not sure I understand distinction from what -- but surely github workflow runs already have some unique ID in their environment that you can use (and that presumably would match some value one can see in the UI).